lib/open_api_spex/test/test_assertions.ex

```diff
@@ -75,4 +122,60 @@ defmodule OpenApiSpex.TestAssertions do
   def assert_request_schema(value, schema_title, api_spec = %OpenApi{}) do
     assert_schema(value, schema_title, api_spec, :write)
   end
+
+  @doc """
+  Asserts that the response body conforms to the response schema for the operation with id `operation_id`.
+  """
+  @spec assert_operation_response(String.t(), Plug.Conn.t()) ::
+          no_return | term
+  def assert_operation_response(operation_id, conn) do
+    {spec, operation_lookup} = PutApiSpec.get_spec_and_operation_lookup(conn)
+
+    case operation_lookup[operation_id] do
+      nil ->
+        flunk(
+          "Failed to resolve schema. Unable to find a response for operation_id: #{operation_id} for response status code: #{conn.status}"
+        )
+
+      operation ->
+        validate_operation_response(conn, operation, spec)
+    end
+  end
+
+  @spec validate_operation_response(
+          Plug.Conn.t(),
+          Operation.t(),
+          OpenApi.t()
+        ) ::
+          term | no_return
+  defp validate_operation_response(conn, %Operation{operationId: operation_id} = operation, spec) do
+    content_type = Utils.content_type_from_header(conn)
+
+    resolved_schema =
+      get_in(operation, [
+        Access.key!(:responses),
+        Access.key!(conn.status),
+        Access.key!(:content),
+        content_type,
+        Access.key!(:schema)
+      ])
+
+    if resolved_schema == nil do
+      flunk(
+        "Failed to resolve schema! Unable to find a response for operation_id: #{operation_id} for response status code: #{conn.status} and content type #{content_type}"
+      )
+    end
+
+    body =
+      case content_type do
+        "application/json" -> Jason.decode!(conn.resp_body)
+        _ -> conn.resp_body
+      end
```

Comment:
This is brittle, but I'm not too sure of what we should do here. There are quite a few content types that should end up calling `Jason.decode()` and I'm not too sure on how you'd let a consumer pass in a collection of json content types that should be encoded (e.g. application/json, application/vnd.api+json, etc).

Reply:
+1 for `application/vnd.api+json`

Reply:
@duntonr I'll fix this with a regex to support the full list of content types. Will update shortly 🎉

Reply:
Updated to reflect the regex used elsewhere to match basically everything with application json in it.